### PR TITLE
Cicd

### PR DESCRIPTION
**Pull request by marshall-mcmullen**
This new `cicd` code to build foundation for automatically merging code from `develop` -> `main` after a PR is merged. This will be extended, modified, and enhanced later as needed for additional workflows we want to support.

This also changes ebash itself to use this new `cicd` code for its own internal CI/CD releases and versioning as well. No more manual merging or tagging. Why didn't I do this YEARS ago. 

Also changed `master` -> `main`

Also bumped supported versions of alpine to the latest.

### CODE REVIEW

File: .github/workflows/Pipeline.yml [hunk 1]
```diff
@@ -139,3 +139,25 @@ jobs:
...
+  #--------------------------------------------------------------------------------------------------------------------
+  #
+  # Release
+  #
+  #--------------------------------------------------------------------------------------------------------------------
+  release:
+    name: "Release"
+    needs: [Linux, MacOS]
+    runs-on: ubuntu-latest
+    if: success() && github.ref == 'refs/heads/develop'
```
Comment (marshall-mcmullen) [inline .github/workflows/Pipeline.yml:152]: This part is important. This means we will only merge code from `develop` -> `main` if all the builds passed and we're on the `develop` branch.

File: tests/cicd.etest [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+	  "repo_slug": "etest",
+	  "series": "1.0",
+	  "version": "1.0.0.0",
+	  "version_tag": "v1.0.0.0",
+	  "version_tag_next": "v1.0.0.1"
+	}
+	EOF
+
+    einfo "Expect"
+    #jq . expect.json
```
Comment (thecodenomad) [inline tests/cicd.etest:194]: Did you mean to leave this commented out?

**Review by thecodenomad — APPROVED** [review]
Just had that one commented line in the test, but otherwise this looks awesome!

File: tests/cicd.etest (hunk 2 above)
Comment (marshall-mcmullen) [reply tests/cicd.etest:194]: I did not. I'll fix that... but don't want to re-do 3 hours of tests, so I'll merge it and then fix it.

**Pull request merged**